x-pack/plugins/viz_editor/indexpattern_datasource_plugin/indexpattern_datasource_plugin.tsx

```diff
@@ -4,47 +4,75 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-// @ts-ignore
-import { EuiSuperSelect } from '@elastic/eui';
-import zipObject from 'lodash-es/zipObject';
+import { EuiComboBox } from '@elastic/eui';
 import React, { useEffect, useState } from 'react';
-import { IndexPattern } from 'ui/index_patterns';
 import { FieldListPanel } from '../public/common/components/field_list_panel';
-import { VisModel } from '../public/common/lib';
+import { Datasource, VisModel } from '../public/common/lib';
 import { DatasourcePlugin, PanelComponentProps } from '../public/datasource_plugin_registry';
 import { getIndexPatterns } from './index_patterns';
 
+interface DataPanelState {
+  indexPatterns: Datasource[];
+  selectionTitle: string;
+}
+
+function getSelectedIndexPattern({
+  indexPatterns,
+  selectionTitle,
+}: DataPanelState): Datasource | null {
+  const selected = indexPatterns.find(({ title }) => title === selectionTitle);
```

Comment:
Should this be by title, or by ID? Tittle seems brittle to me, but I may be wrong?

Reply:
I think it won't make a difference in practice- both values are unique